ProjectSidewalk/SidewalkWebpage
# Resolves #1382 / partially resolves #1394: batches validation mission labels by label type

Partial fix for #1394. 
Resolves #1382.

I am planning on creating a separate pull request to prioritize which validation labels to get, because I think that might be a harder problem than anticipated. This pull request has also been bigger than I imagined, so I think there's enough to test here!

**Changes**: 
* Prevents users from validating their own labels
* Prevents users from validating labels more than once.
* Translate queries to get validation labels to use Slick syntax.
* Displays a message to the user (see screenshot) that says there are no more missions left when there are not enough labels to validate.
![image](https://user-images.githubusercontent.com/25534091/52552003-8e28ed00-2d93-11e9-81c1-bc50fb8af94b.png)
* Updated screen for when there are no more missions available:
![image](https://user-images.githubusercontent.com/25534091/53224677-71f63d00-362a-11e9-8c2d-090db6312442.png)
* Send out Curb Ramp, No Curb Ramp, Surface Problem and Obstacle in Path labels before No Sidewalk labels
* Check to see there are no more missions left for all label types before sending out the "No more missions" screen
* Fixed skip button so it only returns labels belonging to the current mission’s label type id

**Testing Instructions:**
If you run out of labels, you might want to open an incognito window and use the interface as an anonymous user.

1. [Optional] Reload a new dump of the database. If you have a lot of labels in the database, this PR may be a little difficult to test.
2. Visit the validation page at `0.0.0.0:9000/validate`. Check `sbt` - there should be some `println` statements that print out `user_id`, `labelTypeId`, and `labelsRemaining` to the console. If there are less than 10 labels remaining, you should get a screen that says overlay saying there are no more labels left. If not, you should get a mission.
3. Check that the number of labels is correct by running comparing `labelsRemaining` to the size of the following query:
```
SELECT lb.label_id, ms.user_id
FROM sidewalk.label AS lb,
     sidewalk.gsv_data AS gd,
     sidewalk.mission AS ms
WHERE lb.deleted = false
      AND lb.tutorial = false
      AND lb.label_type_id = <label type id>
      AND gd.gsv_panorama_id = lb.gsv_panorama_id
      AND gd.expired = false
      AND ms.mission_id = lb.mission_id
      AND ms.user_id NOT LIKE <user id>
      AND lb.label_id NOT IN (
            SELECT label_id 
            FROM sidewalk.label_validation AS lv
            WHERE lv.user_id LIKE <user id>
      )
```
As an extra sanity check, you can check the `user_id` column to make sure that your current `user_id` doesn't appear there.

4. When you load the mission page (and receive a mission), run this function. None of the IDs here should be repeated.
```
svv.panoramaContainer.getLabels().forEach(function(element) {
    console.log(element.getOriginalProperty("labelId"));
});
```

5. Complete several validation missions. When you run out of labels, the "no more missions" overlay should appear.
6. Other ways you can think to test!
